File: game-server/src/main/java/com/kingston/mmorpg/game/ai/fsm/Patrol2AttackTransition.java
package com.kingston.mmorpg.game.ai.fsm;

import com.kingston.mmorpg.game.database.user.entity.PlayerEnt;
import com.kingston.mmorpg.game.scene.actor.Creature;

public class Patrol2AttackTransition extends Transition {

	public Patrol2AttackTransition(State from, State to) {
		super(from, to);
	}

	@Override
	public boolean meetCondition(Creature creature) {
		// 如果当前在巡逻状态，且怪物没死，就揍它
		PlayerEnt player = (PlayerEnt) creature;
		Scene scene = player.getScene();

		return !scene.getMonster().isDie();
	}
	
}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/ai/fsm/Patrol2AttackTransition.kt
package org.forfun.mmorpg.game.ai.fsm

import org.forfun.mmmorpg.game.scene.actor.Creature
import org.forfun.mmorpg.game.database.user.entity.PlayerEnt

class Patrol2AttackTransition(from: State, to: State) : Transition(from, to) {

    override fun meetCondition(creature: Creature): Boolean {
        // 如果当前在巡逻状态，且怪物没死，就揍它
        val player = creature as PlayerEnt
        val scene = player.scene

        return !scene?.monster!!.isDie
    }
}
